### ISSUE TITLE
Not able to set a heading level when using WPF textblock control.

### ISSUE THREAD

**Issue by adarshrema** [dotnet/wpf#1825]
.net core version 4.7.2

Description: Accessibility Insights for Windows uses the WPF textblock control in various parts of the application. As heading level cannot be associated to textblock control and exposed to Assistive Technologies, a user using screen reader is unable to understand the logical structure of the screen.

Application: Accessibility Insights for Windows, download from:
https://github.com/microsoft/accessibility-insights-windows/releases/download/v1.1.0933.01/AccessibilityInsights.msi

Environment details:
Application Name: Accessibility Insights for Windows, Version: 1.1.933.1
Windows Version: Windows10
Screen Reader: Narrator, NVDA (Version: 2019.1.1)

Repro steps:
1. Open Accessibility insights for Windows application.
2. 'Inspect Home' screen will appear.
3. Go to left hand side navigation bar and select "Setting" Tab.
4. 'Application' screen will appear.
5. Start screen reader.
6. Navigate to the “Settings” text and listen if is announced as header or not.

Actual behavior:
When screen reader users navigate to the “Settings” text it is not announced as header to them. Similarly, for the texts such as “shortcuts”, “Selection type”, “Other Options”, “Telemetry”.

This issue is observed in both the screen readers i.e. NVDA and Narrator.

Similar issue is observed in the below scenarios:

1) Setting –> What's New – The “What’s new – Current” text is not announced as header.
Open Accessibility insight for Windows application. -> 'Inspect Home' screen will appear. -> Go to left hand side navigation bar and select "Setting" Tab. -> Navigate to What's New tab item and hit enter. -> 'What's New' screen will appear. -> Verify all the elements of 'What's New' screen are accessible and MAS Compliant.
2) Setting –> About - The “Accessibility Insights for Window” text is not announced as a header.
Open Accessibility insight for Windows application. -> 'Inspect Home' screen will appear. -> Go to left hand side navigation bar and select "Setting" Tab. -> Navigate to 'About' tab item and hit enter. -> 'About' screen will appear. -> Verify all the elements of 'About' screen are accessible and MAS Compliant.
3) Setting –> Connection Azure Boards - The “Issue Filling” text is not announced as a header.
Open Accessibility insight for Windows application. -> 'Inspect Home' screen will appear. -> Go to left hand side navigation bar and select "Setting" Tab. -> Navigate to 'Connection' tab item and hit enter. -> Select Azure Boards radio button. -> 'Connection Azure Boards' screen will appear. -> Verify all the elements of 'Connection Azure Boards' screen are accessible and MAS Compliant.

Expected behavior:
The " Settings” text should be announced as a header to the screen reader users.

User Impact:
If a certain text which appears as a header is not announced to the screen reader users as a header then the screen reader users might not understand the structure of the content.

MAS reference: MAS 2.4.6

+ @robgallo for visibility

**Comment by weltkante** [comment]
Its the first I'm hearing of "heading levels" for text, is this a new concept? How does it map in other UI frameworks like WinForms and WinUI, does it work out of the box there? In particular you'll want to coordinate with WinUI since its also XAML based and there is desire to keep concepts aligned instead of drifting apart further.

**Comment by RobGallo** [comment]
Check out https://docs.microsoft.com/en-us/windows/uwp/design/accessibility/landmarks-and-headings. This shows how one can indicate heading levels and landmarks on text in UWP.

**Comment by vedeevi** [comment]
Is this issue got fixed and by when can we expect this release available to developer community... We are also facing this issue and need a fix as it is causing problem to Accessibility challenged users for the apps developed using WPF XAML.

Thanks,
vedeevi.

**Comment by anjalisheel-wpf** [comment]
@adarshrema @vedeevi Changes are there in the latest release, hence marking it as closed.
